providers/dns/liquidweb/liquidweb.go

```diff
@@ -0,0 +1,133 @@
+package liquidweb
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/go-acme/lego/v3/challenge/dns01"
+	"github.com/go-acme/lego/v3/platform/config/env"
+	lw "github.com/liquidweb/liquidweb-go/client"
+	"github.com/liquidweb/liquidweb-go/network"
+)
+
+// Config is used to configure the creation of the DNSProvider
+type Config struct {
+	URL                string
+	Username           string
+	Password           string
+	Zone               string
+	Timeout            time.Duration
+	PollingInterval    time.Duration
+	PropagationTimeout time.Duration
+}
+
+// NewDefaultConfig returns a default configuration for the DNSProvider
+func NewDefaultConfig() *Config {
+	config := &Config{
+		Timeout:         env.GetOrDefaultSecond("LIQUID_WEB_TIMEOUT", 1*time.Minute),
+		PollingInterval: env.GetOrDefaultSecond("LIQUID_WEB_POLLING_INTERVAL", 2*time.Second),
+		PropagationTimeout: env.GetOrDefaultSecond("LIQUID_WEB_PROPAGATION_TIMEOUT",
+
+			2*time.Minute),
+	}
+
+	return config
+}
+
+// DNSProvider is an implementation of the acme.ChallengeProvider interface
+// that uses Liquid Web's REST API to manage TXT records for a domain.
+type DNSProvider struct {
+	config    *Config
+	recordIDs map[string]int
```

Comment:
Are you really sure you need that `map`?

Reply:
The API only accepts an integer ID of the record to delete so it has to be tracked someplace.

Reply:
so you have to add a lock.

Reply:
or use `List` or something else.

Reply:
I might not be looking in the right place but it couldn't see why a lock would be necessary. It looks like the DNS solving mechanism submits them all at once using a standard loop.

Reply:
Same for cleanup

Reply:
a map is not thread safe, lego can be used as a lib, so you need to add a lock if someone by example uses lego inside goroutines .

Reply:
Forgot about the lib use case, updating.